internal/appconfig/validation.go

```diff
@@ -1,14 +1,53 @@
 package appconfig
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"reflect"
 	"strings"
 
 	"github.com/go-playground/validator/v10"
+	"github.com/superfly/flyctl/client"
+	"github.com/superfly/flyctl/terminal"
 )
 
-func (c *Config) Validate() (err error) {
+func (cfg *Config) Validate(ctx context.Context) (err error) {
+	appNameFromContext := NameFromContext(ctx)
+
+	apiClient := client.FromContext(ctx).API()
+	appCompact, err := apiClient.GetAppCompact(ctx, appNameFromContext)
+
+	if err != nil {
+		return err
+	}
+
+	if appCompact.PlatformVersion == MachinesPlatform {
+		return cfg.validateLocally()
```

Comment:
Look at `internal/command/config/validate.go`  much of it could be moved or replaced by this. 

Also there are differences on how we validate machines configs.

Reply:
my take would be to validation into `Config.Validate()`, and let it returns enough data so runValidate can render the errors if any. Definitively not make runValidate public.